core/modules/admin_bar/admin_bar.inc

```diff
@@ -816,6 +843,11 @@ function admin_bar_theme_settings() {
   $form['components']['admin_bar.page']['#description'] = t('Links to edit items that make up the current page (e.g. layout, theme).');
   $form['components']['admin_bar.users']['#description'] = t('A count of the people currently logged into the site.');
   $form['components']['admin_bar.account']['#description'] = t('Links to view and log out of the current user account.');
+  $language_switcher_description = t('Links to switch the language of the user interface.');
+  if (!(module_exists('locale'))) {
+    $language_switcher_description .= ' ' . t('This option is available only when the <a href="@link">Locale module</a> is enabled.', array('@link' => url('admin/modules', array('query' => array('search' => 'locale')))));
```

Comment:
@herbdool this here needs to account for when the user doesn't have the permission to manage modules on the site. In that situation, the `<a href="@link">Locale module</a>` thing needs to NOT be rendered as a link (otherwise people will get "access denied").